pybalance/utils/balance_calculators.py

```diff
@@ -629,25 +648,25 @@ def _to_array(self, candidate_populations):
     def _to_list(self, candidate_populations):
         return self.balance_calculator._to_list(candidate_populations)
 
-    def distance(self, candidate_populations):
+    def distance(self, pool_subsets, target_subsets=None):
         # If the user passes a pandas DataFrame, the DataFrame is assumed to
         # represent feature data for only one population and no need for
         # batching; just pass on to the base class. Otherwise, assume
         # candidate_populations refers to patient indices.
-        if isinstance(candidate_populations, pd.DataFrame):
-            return self.balance_calculator.distance(candidate_populations)
+        if isinstance(pool_subsets, pd.DataFrame):
```

Comment:
you're not checking for type of target_subsets? assumed to be the same type as pool_subsets?

Reply:
fixed